### ISSUE TITLE
[Feature]: Add an option to enable only text options in the default context menus

### ISSUE THREAD

**Issue by aquinn39** [MicrosoftEdge/WebView2Feedback#4189]
### Describe the feature/enhancement you need

Currently, there exists an option called AreDefaultContextMenusEnabled which can be set to true or false to enable or disable default context menus in the WebView2. The problem is, often it is desirable to disable various in-built context menu options such as Save as, web capture, copy link, open in new window, etc. while still enabling text options such as Cut, Copy and Paste - in fact, this is the default behaviour of the Microsoft Edge based UWP WebView - it shows only text options in its context menus.

UWP WebView context menu with a link selected:
![image](https://github.com/MicrosoftEdge/WebView2Feedback/assets/43898231/c5f849ce-1289-40f4-b7e5-a76b3d681e96)

WebView2 context menu with a link selected:
![image](https://github.com/MicrosoftEdge/WebView2Feedback/assets/43898231/6da72e6d-5661-4182-b848-6473f9f17bb0)

### The scenario/use case where you would use this feature

This would be useful as in many situations that require a web view, developers want to remove normal browsers items such as the option to open links in new windows, use web capture, etc. but still allow users to copy and paste text from the context menu like the UWP WebView. An example of this could be an in app help manual where these extra options do not make sense but the user may still want to copy text (for example, the user manual might have example input data for the app that the user may want to copy and past into the app). Microsoft's own Get Help app might find this feature useful as well as it disables the default WebView2 context menus but there is a search text box which currently has no context menu and as a user I think it would be beneficial to have a Cut, Copy Paste context menu for that text box.

### How important is this request to you?

Nice to have. There are other ways to tackle this, but having official API support would be beneficial.

### Suggested implementation

At the moment, it is possible to customise the default context menus using the items' names to identify them, however, as far as I know, there is no official documentation that describes the names of the default context menu items and so, as a developer, I am unsure if they might change in the future or if we should be using them. Additionally, since this is such a common scenario, I think it would be beneficial for this to be an option that can be opted into, as that would be much quicker than customising the menu manually. This could also help with migrating from the old UWP WebView, as, when enabled, it would match the context menu behaviour to that of the old UWP WebView. Also, on the topic of the names, see this issue: https://github.com/MicrosoftEdge/WebView2Feedback/issues/3297.

**Comment by victorhuangwq** [comment]
You are currently able to create your own custom context menu using the context menu API:
https://learn.microsoft.com/en-us/microsoft-edge/webview2/how-to/context-menus?tabs=csharp

If you are specifically asking about removing one or few items from the default browser context menu, that would not be possible. In general, WebView2 provides either an API for users to recreate their own custom version of a browser behavior, or just select the original browser behavior.

**Comment by aquinn39** [comment]
> You are currently able to create your own custom context menu using the context menu API:
> 
> https://learn.microsoft.com/en-us/microsoft-edge/webview2/how-to/context-menus?tabs=csharp
> 
> 
> 
> If you are specifically asking about removing one or few items from the default browser context menu, that would not be possible. In general, WebView2 provides either an API for users to recreate their own custom version of a browser behavior, or just select the original browser behavior. 

Yes, as I said I am aware that you can customise the context menu. The main problem is that there is no way to just show text based context menu items, something that is commonly desired and was the behaviour with the old UWP WebView. You can manually customise the context menu to just show items like cut, copy and paste but that requires knowing the names of those items, which are not documented and could changed. Additionally, it would be good to have a property or function for this as it is such a common scenario. I've noticed that many developers resort to turning off the browser context menu instead of bothering to manually customise the menu, which can result in a reduced user experience when the user wants to copy text over Edge legacy's UWP WebView.

Basically, I think it's common that developers would want just the text based options in the context menus and no other options, so an option for this would be nice or at least it would be good if the names of the built-in context menu items could be documented somewhere.

**Comment by victorhuangwq** [comment]
We have a work item to track the documentation of those context menu items, but as of now, @ajtruckle has opened an issue with the names of those items: https://github.com/MicrosoftEdge/WebView2Feedback/issues/3297
